### PR TITLE
Add WebMentions and basic IndieWeb µFormats v2 support (h-card, h-entry)

### LINKED ISSUE #91: IndieWeb support
Hi, I just found out about this project, and it looks like it would do well with some sort of [IndieWeb](https://indieweb.org/) integration.

Briefly, IndieWeb is a set of ideals and protocols that websites can use to follow each other and send each other reples — sound familiar? :) 

The specific things that I'd recommend implementing:

* [h-feed/h-entry/h-card](https://indieweb.org/microformats2) — microformats that are used to provide semantic/sharing information (authorship, post text, etc.); publishing this is as simple as decorating DOM elements with a handful of CSS classes, and receiving it is straightforward with a number of existing parser libraries (including https://github.com/willnorris/microformats for golang)
* [Webmention](https://indieweb.org/webmention) — a mechanism for sending and receiving replies/likes/bookmarks/RSVPs/etc. (which is built on top of the microformats, h-entry and h-card in particular)
* [WebSub](https://indieweb.org/WebSub) — push notifications for subscription updates (useful for RSS/atom readers as well)

Also it isn't clear to me as to whether you have support for RSS/Atom feeds, but if not, it'd be helpful to provide Atom so that other things can interoperate as well. Atom even has support for push (via WebSub) and boosting/reposting content!

Anyway, welcome to the decentralized web neighborhood. I hope to see more from this project. :)

### PR DESCRIPTION
**Pull request by prologic**
##### Summary

ssia

##### Component Name

- area/backend
- area/web

##### Test Plan

Tested estensivally locally and on two test pods.

##### Additional Information

Closes #91

### CODE REVIEW

**Comment by prologic** [comment]
cc @fluffy-critter

File: internal/handlers.go [hunk 1]
```diff
@@ -560,62 +575,68 @@ func (s *Server) PostHandler() httprouter.Handle {
...
 			s.render("error", w, ctx)
 			return
 		}
 
+		// WebMentions ...
+		for _, twter := range twt.Mentions() {
+			WebMention(
+				twter.URL,
+				fmt.Sprintf("%s/twt/%s", s.config.BaseURL, twt.Hash()),
+			)
```
Comment (prologic) [inline internal/handlers.go:612]: TODO: Do these in the background...

File: internal/utils.go [hunk 2]
```diff
@@ -77,6 +77,23 @@ var (
...
+
+	userURL := UserURL(url)
+	username := filepath.Base(userURL)
+
+	return db.GetUser(username)
+}
+
+func WebMention(target, source string) {
+	targetURL, _ := url.Parse(target)
+	sourceURL, _ := url.Parse(source)
```
Comment (prologic) [inline internal/utils.go:93]: TODO:

- [ ] Check for errors

File: internal/templates/_partials.html [hunk 3]
```diff
@@ -52,23 +52,23 @@
       {{ template "pager" . }}
       
       {{ range .Twts }}
-        <article id="{{ .Hash }}">
+        <article id="{{ .Hash }}" class="h-entry">
```
[the remark this answers is not in the document]
Comment (prologic) [reply internal/templates/_partials.html:55]: Done. Please check the latest commit. Thank you 👍

File: internal/templates/_partials.html [hunk 4]
```diff
@@ -92,7 +92,7 @@ <h3>
               </h3>
             {{ end }}
             <footer>
-              <a href="/twt/{{ .Hash }}">
+              <a class="u-url" href="/twt/{{ .Hash }}">
                 <time datetime="{{ .Created | date "2006-01-02T15:04:05Z07:00" }}">
```
Comment (fluffy-critter) [inline internal/templates/_partials.html:96]: Oh, while you're at it, there's also a `dt-published` class that should be added to this `<time>`. See http://microformats.org/wiki/h-entry for all of the fun metadata fields that are available. :)

Comment (prologic) [reply internal/templates/_partials.html]: Done

**Comment by prologic** [comment]
@fluffy-critter Btw adding `h-feed` causes my webmention processor ro fail and panic. I _assume_ that I now have to adjust the processor to firstly look for the `h-feed`, then look inside the children of this `microformats.MicroFormat` struct for the first `h-entry`?

**Comment by fluffy-critter** [comment]
The `h-feed` class should only be on things that are feeds; pages that have just a single entry should only have the `h-entry`.

That said, the webmention endpoint should be validating that the source page has a link to the target page, and should be using the associated `h-entry` to get the metadata for the link. How you do that is up to you but I would suggest looking at all elements that have an `h-entry` class and use the first one that contains the link. I don't know what mf2 library you're using but in mf2py you can limit the query to certain item types.

**Comment by prologic** [comment]
@JsAppNinja Thanks for the approval! If you have some feedback (_noticed your 👎 on some comments_) please speak up 😀

**Comment by prologic** [comment]
Okay I'm merging this now...

In about ~5mins all pods will get updated

**Pull request merged**